spec/api/commentable_spec.rb

```diff
@@ -125,11 +125,11 @@ def test_unicode_data(text)
       subject { post "/api/v1/#{commentable_id}/threads", parameters }
 
       shared_examples_for 'CommentThread creation API' do |context='course'|
-        it 'creates a new CommentThread' do
+        it 'creates a new CommentThread and marks it as read for owner user' do
           expect(CommentThread.count).to eq 0
 
           body = parse(subject.body)
-          expect(body).to include('read' => false,
+          expect(body).to include('read' => true,
```

Comment:
Note that I was just keeping up with email but haven't looked at any code changes.  I'll do that when we return tomorrow.

> On Sep 5, 2016, at 4:02 AM, wajeeha-khalid <notifications@github.com> wrote:
> 
> In spec/api/commentable_spec.rb:
> 
> >            expect(CommentThread.count).to eq 0
> >  
> >            body = parse(subject.body)
> > -          expect(body).to include('read' => false,
> > +          expect(body).to include('read' => true,
> @robrap There are cases (comment POST endpoints); where the returned hash is of comment and not thread. In such cases, we need to make another call to get thread details to assert "read" status.
> 
> I have updated test_thread_marked_as_read in a way that it will get thread details via GET api and assert read=true. It seems like a better approach instead of directly calling on db values.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>